rgbd_image_buffer/src/image_buffer.cpp

```diff
@@ -46,7 +49,7 @@ void ImageBuffer::initialize(const std::string& topic, const std::string& root_f
 
 // ----------------------------------------------------------------------------------------------------
 
-bool ImageBuffer::waitForRecentImage(rgbd::ImageConstPtr& image, geo::Pose3D& sensor_pose, double timeout_sec)
+bool ImageBuffer::waitForRecentImage(rgbd::ImageConstPtr& image, geo::Pose3D& sensor_pose, double timeout_sec, uint timeout_tries)
```

Comment:
Sometimes you may want a number of tries but I think more often you want to specify a frequency directly.

I would suggest to split this in two functions. One which takes timeout+freq and one which takes timeout_tries and calls the former based on the calculated freq. 

I think that would make for a better separation of concerns.